[Bug]: pnpm install fail in devcontainer

### Describe the bug

Dev container를 재생성할 경우 link 문제로 `pnpm install`이 제대로 동작하지 않는 경우가 많습니다.
주로 macOS에서 docker mount를 사용할 때 자주 일어나는 문제인 것 같습니다.
https://github.com/pnpm/pnpm/issues/7024

### Solution

`store-dir` 옵션을 mount되지 않은 container directory로 변경합니다.
https://github.com/pnpm/pnpm/issues/3952#issuecomment-1262136483

명령어로 설정하는 것보다는 container 내부에서 환경 변수를 설정하는 것이 좋아보입니다.
https://pnpm.io/npmrc#store-dir
https://github.com/pnpm/pnpm/issues/7024

### Reproduction

1. dev container 삭제
2. container 다시 build
3. `pnpm install`


### Logs

```bash
$ pnpm install
Scope: all 3 workspace projects
Lockfile is up to date, resolution step is skipped
Packages: +1670
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 ERR_PNPM_LINKING_FAILED  Error: ENOENT: no such file or directory, copyfile '/workspace/.pnpm-store/v3/files/4e/e1c88f8c3f4e4cd34cb6c00339bf9d6d036ff4ade3af49e871cc8966b84c729d8b75492acc6413c9a664ac00a57958223ac13c4229da8c62ebe6a53e4f783f' -> '/workspace/node_modules/.pnpm/@aws-sdk+types@3.418.0/node_modules/@aws-sdk/types_tmp_2790/dist-es/connection.js'
```

### System Info

```

                 -/+:.          dotol@choejaemin-ui-MacBookAir
                :++++.          OS: 64bit macOS
               /+++/.           Kernel: arm64 Darwin 22.6.0
       .:-::- .+/:-``.::-       Uptime: 19d 19h 38m
    .:/++++++/::::/++++++/:`    Packages: 163
  .:///////////////////////:`   Shell: zsh 5.9
  ////////////////////////`     Resolution: 2560x1664, 2560x1440
 -+++++++++++++++++++++++`      DE: Aqua
 /++++++++++++++++++++++/       WM: Quartz Compositor
 /sssssssssssssssssssssss.      WM Theme: Blue
 :ssssssssssssssssssssssss-     Disk: 282G / 494G (60%)
  osssssssssssssssssssssssso/`  CPU: Apple M2
  `syyyyyyyyyyyyyyyyyyyyyyyy+`  GPU: Apple M2
   `ossssssssssssssssssssss/    RAM: 1284MiB / 16384MiB
     :ooooooooooooooooooo+.
      `:+oo+/:-..-:/+o+/-

```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/skkuding/next/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/skkuding/next/blob/main/CONTRIBUTING.md)
- [X] Check that there isn't already an issue

--- a/.npmrc
+++ b/.npmrc
@@ -1,1 +1,2 @@
 auto-install-peers=true
+package-import-method=clone-or-copy

